egps/egp-15/egp-15-retro.ts

```diff
@@ -0,0 +1,121 @@
+import { BigNumber, BytesLike, ethers } from 'ethers'
+import { LedgerSigner } from '@ethersproject/hardware-wallets'
+
+// Artifacts
+import timelockData from '../../artifacts/council/contracts/features/Timelock.sol/Timelock.json'
+import treasuryData from '../../artifacts/council/contracts/features/Treasury.sol/Treasury.json'
+import lockingVaultData from '../../artifacts/council/contracts/vaults/LockingVault.sol/LockingVault.json'
+import { CoreVoting__factory } from '../../typechain/council/factories/CoreVoting__factory'
+
+// Helpers
+import * as addresses from '../helpers/addresses'
+import { createCallHash } from '../helpers/hashing'
+
+async function proposal() {
+  // Setup your signer
+  // Using a ledger, connect to metamask
+  const providerUrl = ''
+  const provider = new ethers.providers.JsonRpcProvider(providerUrl)
+  const signer = new LedgerSigner(provider, undefined, "m/44'/60'/3'/0/0")
+
+  // Setup your interfaces
+  const timelockInterface = new ethers.utils.Interface(timelockData.abi) // TimeLock is like sudo, you always need it.
+  const treasuryInterface = new ethers.utils.Interface(treasuryData.abi) // TimeLock is like sudo, you always need it.
+  const lockingVaultInterface = new ethers.utils.Interface(lockingVaultData.abi)
+
+  // New Params
+  // @TODO - fill in.
+
+  // Connect the signer to the coreVotingContract, this is where your proposals will fed into.
+  const coreVoting = CoreVoting__factory.connect(addresses.CoreVoting, signer)
+
+  // --- main egp logic ---
+  /**
+   * The Timelock contract has to call deposit() on the locking vault contract through the Treasury contract...
+   * 1. Generate calldata for deposit() call on locking vault
+   * 2. Generate calldata for genericCall() call on treasury
+   * 3. Generate callhash for timelock
+   */
+
+  /** Generate calldata for locking vault deposit */
+  const retroAmount = ethers.utils.parseEther('19531.25')
```

Comment:
this parses to 18 decimals ? Is that how ELFI is setup?

Reply:
[etherscan](https://etherscan.io/token/0x4da34f8264cb33a5c9f17081b9ef5ff6091116f4) is showing 18 decimals for ELFI so I think this is okay